configure.ac

```diff
@@ -7551,6 +7551,29 @@ WITH_SAVE_ENV([
   ])
 ])
 
+WITH_SAVE_ENV([
+  LIBS="$LIBS $OPENSSL_LIBS"
+  CFLAGS="$CFLAGS $OPENSSL_INCLUDES"
+  LDFLAGS="$LDFLAGS $OPENSSL_LDFLAGS $OPENSSL_LDFLAGS_RPATH"
+
+AC_CHECK_HEADERS([openssl/ech.h], [have_openssl_ech_h=yes], [have_openssl_ech_h=no])
+if test "$have_openssl_ech_h" = "yes"; then
+    AC_MSG_CHECKING([for OpenSSL ECH support])
+    AC_CHECK_DECLS([SSL_ECH_STATUS_GREASE_ECH],
+                   [AC_DEFINE([OPENSSL_ECH], [1],
```

Comment:
Can we use `HAVE_OPENSSL_ECH` naming as for other macros please? See the AC func PY_CHECK_FUNC.

Reply:
I've updated this to use PY_CHECK_FUNC but honestly I have no idea what I'm doing with autoconf and I could not find any documentation on correct usage of PY_CHECK_FUNC so it may just be working by chance.